### PR TITLE
[foder] Introduce file loader module

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit introduces foder to compiler

Draft: #2201 
Related : #2166

ONE-DCO-1.0-ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: compiler/foder/include/foder/FileLoader.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    // reserve capacity
+    _data.reserve(_size);
+
+    // read the data
+    file.read(_data.data(), _size);
+    if (file.fail())
+      return nullptr;
+
+    return _data.data();
+  }
```
Comment (parjong) [inline compiler/foder/include/foder/FileLoader.h:57]: Do you need to keep ``DataBuffer`` insider this class? Why don't you simply return ``std::vector<char>`` and throws an exception on failure?

The run-time behavior would be almost same as this implementation due to RVO (Return-value optimization).

**Comment by mhs4670go** [comment]
@seanshpark @parjong PTAL:)

**Comment by mhs4670go** [comment]
@seanshpark sorry, I've changed comments a little bit. PTAL

**Comment by seanshpark** [comment]
maybe fix typo title ? `loder ` :)

File: compiler/foder/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# foder
+
+_foder_ is a header only library that loads files.
+
+## Example
+
+```cpp
+foder::FileLoader fileloader{input_path};
+
+const char *data = fileloader.load();
```
Comment (seanshpark) [inline compiler/foder/README.md:10]: maybe update this example?

File: compiler/foder/include/foder/FileLoader.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+public:
+  explicit FileLoader(const std::string &path) : _path(path) {}
+
+public:
+  FileLoader(const FileLoader &) = delete;
+  FileLoader(FileLoader &&) = delete;
+
+public:
+  const DataBuffer load(void)
```
Comment (parjong) [inline compiler/foder/include/foder/FileLoader.h:36]: (Optional) const has no effect in this case ;-D

File: compiler/foder/include/foder/FileLoader.h [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      throw std::runtime_error("Couldn't read file.");
+
+    return data;
+  }
+
+  inline std::streampos size(void) { return _size; }
+
+private:
+  const std::string _path;
+  std::streampos _size;
```
Comment (parjong) [inline compiler/foder/include/foder/FileLoader.h:63]: (Optional) This filed is redundant.

File: compiler/foder/include/foder/FileLoader.h (hunk 3 above)
Comment (parjong) [reply compiler/foder/include/foder/FileLoader.h:36]: ``load`` method now can be a ``const`` method ;-D

File: compiler/foder/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+_foder_ is a header only library that loads files.
+
+## Example
+
+```cpp
+foder::FileLoader fileloader{input_path};
+
+std::vector<char> data = fileloader.load();
+
+const circle::Model *circle_model = circle::GetModel(data.data());
```
Comment (parjong) [inline compiler/foder/README.md:12]: (Optional) This may confuse users. A _foder_ user is not specific to _circle_ developers ;-)

**Pull request merged**